scripts/tests/java/im_test.py

```diff
@@ -0,0 +1,120 @@
+        logging.info(f"Execute: {java_command}")
+        java_process = subprocess.Popen(
+            java_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        DumpProgramOutputToQueue(self.thread_list, Fore.GREEN + "JAVA " + Style.RESET_ALL, java_process, self.queue)
+        return java_process.wait()
+
+    def TestCmdOnnetworkLongImRead(self, nodeid, setuppin, discriminator, timeout):
+        java_command = self.command + ['im', 'onnetwork-long-im-read', nodeid, setuppin, discriminator, timeout]
+        logging.info(f"Execute: {java_command}")
+        java_process = subprocess.Popen(
+            java_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        DumpProgramOutputToQueue(self.thread_list, Fore.GREEN + "JAVA " + Style.RESET_ALL, java_process, self.queue)
+        return java_process.wait()
+
+    def TestCmdOnnetworkLongImSubscribe(self, nodeid, setuppin, discriminator, timeout):
+        java_command = self.command + ['im', 'onnetwork-long-im-subscribe', nodeid, setuppin, discriminator, timeout]
+        logging.info(f"Execute: {java_command}")
+        java_process = subprocess.Popen(
+            java_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        DumpProgramOutputToQueue(self.thread_list, Fore.GREEN + "JAVA " + Style.RESET_ALL, java_process, self.queue)
+        return java_process.wait()
+
+    def RunTest(self):
+        if self.command_name == 'onnetwork-long-im-invoke':
+            logging.info("Testing pairing onnetwork-long-im-invoke")
+            code = self.TestCmdOnnetworkLongImInvoke(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-invoke failed with error {code}")
+        elif self.command_name == 'onnetwork-long-im-write':
+            logging.info("Testing pairing onnetwork-long-im-write")
+            code = self.TestCmdOnnetworkLongImWrite(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-write failed with error {code}")
+        elif self.command_name == 'onnetwork-long-im-read':
+            logging.info("Testing pairing onnetwork-long-im-read")
+            code = self.TestCmdOnnetworkLongImRead(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-read failed with error {code}")
+        elif self.command_name == 'onnetwork-long-im-subscribe':
+            logging.info("Testing pairing onnetwork-long-im-subscribe")
+            code = self.TestCmdOnnetworkLongImSubscribe(self.nodeid, self.setup_pin_code, self.discriminator, self.timeout)
+            if code != 0:
+                raise Exception(f"Testing pairing onnetwork-long-im-subscribe failed with error {code}")
```

Comment:
I am still confused with this implementation. It looks the IM cluster and pairing cluster are mangled together in test "Testing pairing onnetwork-long-im-subscribe"

From the implementation side, I saw all IM commands are registered to the 'im' cluster

`  commandManager.register("im", getImCommands(controller, credentialsIssuer))`

But implemented as a 'pairing' command

```
class PairOnNetworkLongImInvokeCommand(
  controller: ChipDeviceController, credsIssue: CredentialsIssuer?
) : PairingCommand
```

Reply:
yes, in current test, we always starts from fresh state, so we need pair first, then run im commands